NoSQL/database_design.txt

```diff
@@ -0,0 +1,105 @@
+1) Collection users:
+
+Design of single entry for user:
+
+{
+	_id : <user_id>,
+
+	credentials : {
+		email : <email_id>,
+		password : <password>
+	},
+
+	profile_details : {
+		name : {
+			first_name : <first_name>,
+			last_name : <last_name>
+		},
+
+		date_of_birth : <date>,
+
+		profile_picture : <url>,
+
+		contact : <contact>,
+
+		place : <city-state>,
+
+		<other_profile_details>
+	},
+
+	roles : {
+		position : <user/admin/editor/superadmin>,
+
+		can_update : <boolean>,
+
+		can_delete : <boolean>
+	},
+
+	activity : {
+		post : [
+			{
+				post_id : <post_id>,
+
+				timestamp : <date_time>
+			},
+
+			<other_entries_for_post_activity>
+		],
+
+		liked : [
+			{
+				post_id : <post_id>,
+
+				timestamp : <date_time>
+			},
+
+			<other_entries_for_like_activity>
+		]
+	}
+}
+
+
+
+2) Collection posts:
+
+Design of single entry for post:
+
+{
+	_id : <post_id>,
+
+	user_id : <user_id>
+
+	post : {
```

Comment:
you don't need this sub document. The collection itself is posts, and all of the details would be required in a single fetch.